From: Koichiro Den <koichiro.den@canonical.com>
Date: Fri, 7 Feb 2025 01:22:16 +0900
Subject: [PATCH] Revert "btrfs: avoid monopolizing a core when activating a
 swap file"
References: bsc#1234429
Patch-mainline: 6.12.14
Git-commit: 118b41e78b0ad47f257b0dfbcb7956f3b0b6cf7d

This reverts commit 9f372e86b9bd1914df58c8f6e30939b7a224c6b0.

The backport for linux-6.12.y, commit 9f372e86b9bd ("btrfs: avoid
monopolizing a core when activating a swap file"), inserted
cond_resched() in the wrong location.

Revert it now; a subsequent commit will re-backport the original patch.

Fixes: 9f372e86b9bd ("btrfs: avoid monopolizing a core when activating a swap file") # linux-6.12.y
Signed-off-by: Koichiro Den <koichiro.den@canonical.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Robert Frohl <rfrohl@suse.com>
---
 fs/btrfs/inode.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/fs/btrfs/inode.c b/fs/btrfs/inode.c
index 9d9ce308488d..197ba9be9718 100644
--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -7200,8 +7200,6 @@ noinline int can_nocow_extent(struct inode *inode, u64 offset, u64 *len,
 			ret = -EAGAIN;
 			goto out;
 		}
-
-		cond_resched();
 	}
 
 	if (file_extent)
-- 
2.48.0

